### PR TITLE
Implement latest validator convention.

### LINKED ISSUE #46: Change validator return value
We have agreed to adopt the convention that all validators need to return {valid: <bool>, error: <error>}.  This module an the existing validators need to be updated to use that pattern.

### PR DESCRIPTION
**Pull request by mattcollier**
Fixes #46.

This will be part of a major release.

### CODE REVIEW

File: lib/validator.js [hunk 1]
```diff
@@ -97,37 +97,44 @@ async function _runValidator(
...
+  let result;
   try {
-    await validatorApi.validate(
+    result = await validatorApi.validate(
       {validatorInput, ledgerConfig, ledgerNode, validatorConfig});
   } catch(e) {
-    report.error = serializeError(e);
+    // if a validator throws, it is an uncaught exception. Serialize the error
+    // so that it will appear in the validator report for debugging purposes
+    result = {valid: false, error: serializeError(e)};
```
Comment (dlongley) [inline lib/validator.js]: Doesn't `serializeError` create a JSON object/string? Don't we want `error` to just be an `Error` instance?

File: lib/validator.js [hunk 2]
```diff
@@ -97,37 +97,44 @@ async function _runValidator(
...
   }
 
-  report.validated = !report.error;
+  report.valid = result.valid;
+  report.error = result.error;
```
Comment (dlongley) [inline lib/validator.js]: Isn't this where `serializeError` should be used (for the `report`)?

**Comment by mattcollier** [comment]
@dlongley I believe the serializeError/reporting stuff does what we want.  The key thing that was not happening was that if a validator was throwing in dev due to something like 'can't access foo of undefined' that error was not making it into the report thus making it difficult to debug.

**Comment by dlongley** [comment]
Yes, but we want `result = {valid: <boolean>, error: <Error instance>}` and `report.error = serializeError(result.error)` ... don't we? I think it's weird to not make `error` on the return value an `Error` instance when we do that everywhere else.

**Comment by mattcollier** [comment]
@dlongley your suggestion works as well.  Pushed.

**Pull request merged**